modules/candidate_parameters/jsx/ExternalIdentifier.js

```diff
@@ -0,0 +1,350 @@
+                  key
+                )
+              }
+            />
+            <hr />
+          </div>
+        );
+      }
+    }
+
+    var IDRequired = false;
+    if (this.state.formData.ProjectID) {
+      IDRequired = true;
+    }
+
+    var ProjectRequired = true;
+
+    var alertMessage = "";
+    var alertClass = "alert text-center hide";
+    if (this.state.updateResult) {
+      if (this.state.updateResult === "success") {
+        alertClass = "alert alert-success text-center";
+        alertMessage = "Update Successful!";
+      } else if (this.state.updateResult === "error") {
+        var errorMessage = this.state.errorMessage;
+        alertClass = "alert alert-danger text-center";
+        alertMessage = errorMessage ? errorMessage : "Failed to update!";
+      }
+    }
+
+    return (
+      <div className="row">
+        <FormElement
+          name="externalIdentifier"
+          onSubmit={this.handleSubmit}
+          ref="form"
+          class="col-md-6"
+        >
+          <StaticElement
+            label="PSCID"
+            text={this.state.Data.pscid}
+          />
+          <StaticElement
+            label="DCCID"
+            text={this.state.Data.candID}
+          />
+          <hr />
+          {existingStudiesHTML}
+          <div className={alertClass} role="alert" ref="alert-message">
+            {alertMessage}
+          </div>
+          <SelectElement
+            label="Study"
+            name="ProjectID"
+            options={studyOptions}
+            onUserInput={this.setFormData}
+            ref="ProjectID"
+            disabled={disabled}
+            required={ProjectRequired}
+            value={this.state.formData.ProjectID}
+          />
+          <TextboxElement
+            label="External Study Identifier"
+            name="ExtStudyID"
+            onUserInput={this.setFormData}
+            ref="ExtStudyID"
+            disabled={disabled}
+            required={IDRequired}
+            value={this.state.formData.ExtStudyID}
+          />
+          {addButton}
+        </FormElement>
+      </div>
+    );
+  },
+
+  /**
+   * Handles form submission
+   *
+   * @param {event} e - Form submission event
+   */
+  handleSubmit: function(e) {
+    e.preventDefault();
+    var myFormData = this.state.formData;
+    var self = this;
+    var formData = new FormData();
+    var formRefs = this.refs;
+
+    var existingStudies = this.state.existingStudies;
+    var existingStudy = {};
+
+    for (var key in myFormData) {
+      if (myFormData.hasOwnProperty(key)) {
+        if (myFormData[key] !== "") {
+          existingStudy[key] = myFormData[key];
+          formData.append(key, myFormData[key]);
+        }
+      }
+    }
+
+    formData.append('tab', this.props.tabName);
+    formData.append('candID', this.state.Data.candID);
+
+    existingStudies.push(existingStudy);
+
+    this.setState({
+      existingStudies: existingStudies
+    });
+
+    $.ajax({
```

Comment:
are we moving towards using fetch instead and getting rid of the jQuery dependency? @HenriRabalais @davidblader @zaliqarosli @xlecours ?

I know that's what's happening on the menu filter PRs, and it's probably a good idea to make it universal?

Reply:
agree with using fetch and with removing jQuery dependency in general, however the fade animation in `showAlertMessage` might be tricky/somewhat time consuming to refactor w/o jQuery and i wouldn't necessarily block on that basis at this moment. using fetch is a good start though

Reply:
I'm also undecided on if we should block over the ES6 thing.. it's a useful feature to block over style which might take a lot of effort to fix, but @zaliqarosli's also put a lot of effort into modernizing all of our existing javascript into ES6 and since it's a new class it would be good not to have regressions in that regard.

Reply:
IMO Reactification outranks ES standards since it's blocking so many other things such as using the data provisioners, project permissions, etc.

We can always make a ticket to update this code to ES6 standards. It's a good beginner/summer student task.

Reply:
This isn't Reactification though, this is adding a new tab /feature to a module that's already written in React.

Reply:
showAlertMessage can be replaces with swal which is very easy to implement, and looks very clean in ES6

Reply:
also, any future reactifications should be coded in ES6 anyways

Reply:
I'm not gonna go down that road. thx for the feedback but I'm most probably not gonna have time to refactor anything on the JS side. this PR will most probably be closed

if anyone ever decides to tackle the Candidate params module. keep this PR in mind